galasa-managers-parent/galasa-managers-comms-parent/dev.galasa.mq.manager/src/main/java/dev/galasa/mq/internal/MQManagerImpl.java

```diff
@@ -0,0 +1,253 @@
+    	String qmgrTag = annotation.queueMgrTag();
+    	if(!queueManagers.containsKey(qmgrTag)) {
+    		throw new MqManagerException("Unable to provision queue: " + annotation.name() + " no QMGR found tagged: " + qmgrTag);
+    	}
+    	
+    	//check if the name is specified in the queue or in the CPS
+    	String queueName = annotation.name();
+    	String queueTag = annotation.tag();
+    	
+    	if(queueName.isEmpty() && queueTag.isEmpty())
+    		throw new MqManagerException("Either name or tag must be specified in @Queue annotation");
+    	
+    	if(!queueName.isEmpty() && !queueTag.isEmpty())
+    		throw new MqManagerException("Both name and tag are specified in @Queue annotation, these are mutually exclusive");
+    	
+    	String name = queueName;
+    	if(queueTag.length() > 0){
+    		name = QueueNameForTag.get(queueTag);
+    		if(name == null) {
+    			throw new MqManagerException("No queue tag for " + queueTag);
+    		}
+    	}
+    	
+    	//construct the queue and add it to our list
+        MessageQueueImpl queue = new MessageQueueImpl(name,queueManagers.get(qmgrTag), Boolean.parseBoolean(annotation.archive()), this);
+        this.queues.add(queue);
+        registerAnnotatedField(field, queue);
+        return queue;
+    }
+    
+    @GenerateAnnotatedField(annotation = QueueManager.class)
+    public IMessageQueueManager generateMessageQueueManager(Field field, List<Annotation> annotations) throws MqManagerException{
+    	//obtain the tag for this queue manager
+    	QueueManager annotation = field.getAnnotation(QueueManager.class);
+    	String tag = annotation.queueMgrTag();
+		
+		//obtain the configuration for this queue manager
+		String instanceid = getInstanceForTag(tag);
+		logger.info("Obtaining configuration information for instance: " + instanceid);
+		String host = InstanceHost.get(instanceid);
+		int port = InstancePort.get(instanceid);
+		String channel = InstanceChannelName.get(instanceid);
+		String name = InstanceName.get(instanceid);
+
+		//construct the queue manager and add it to the list
+        MessageQueueManagerImpl qmgr = new MessageQueueManagerImpl(tag, name, host, port, channel, this);
+        this.queueManagers.put(tag, qmgr);
+        registerAnnotatedField(field, qmgr);
+        return qmgr;
+    }
+
+    @Override
+    public void provisionGenerate() throws ManagerException, ResourceUnavailableException {
+    	List<AnnotatedField> annotatedFields = findAnnotatedFields(MqManagerField.class);
+    	
+    	//First generate the queue managers
+    	for(AnnotatedField af : annotatedFields) {
+    		Field f = af.getField();
+    		if(f.getType() == IMessageQueueManager.class) {
+    			QueueManager annotation = f.getAnnotation(QueueManager.class);
+    			if(annotation != null) {
+    				generateMessageQueueManager(f, af.getAnnotations());
+    			}
+    			
+    		}
+    	}
+        generateAnnotatedFields(MqManagerField.class);
+    }
+
+    @Override
```

Comment:
If you are not using this, no need to override it

Reply:
Thanks - done that